build.sh

```diff
@@ -218,15 +218,27 @@ export DOCKER_TAG
 # Run the requested commands.
 
 if [[ -n "$IMAGE_BUILD" ]]; then
-  (set -x; docker compose build build)
+  # Use Docker Bake if available for better caching support, otherwise fall back to compose
+  if command -v docker &> /dev/null && docker buildx version &> /dev/null && [[ -f docker-bake.hcl ]]; then
```

Comment:
You don't need to check for the existence of docker or the HCL file. Assume they exist.